packages/hash/frontend/src/lib/entities.ts

```diff
@@ -467,22 +468,30 @@ export const generateEntityLabel = (
     "shortname",
   ];
 
-  /** @todo refactor the following section to make it more readable */
-  const propertyTypes: { title?: string; propertyTypeId: string }[] =
-    Object.keys(entity.properties).map((propertyTypeId) => ({
-      propertyTypeId,
-      title: (
-        entity as EntityResponse
-      ).entityTypeRootedSubgraph.referencedPropertyTypes
-        .find((item) => item.propertyTypeId.startsWith(propertyTypeId))
-        ?.propertyType.title.toLowerCase(),
-    }));
+  const propertyTypes: { title?: string; propertyTypeBaseUri: string }[] =
+    Object.keys(entity.properties).map((propertyTypeBaseUri) => {
+      const propertyTypeVersions = getPropertyTypesByBaseUri(
+        (entity as EntityResponse).entityTypeRootedSubgraph,
+        propertyTypeBaseUri,
+      );
+
+      /** @todo - pick the latest version? */
```

Comment:
We're going through the `properties` of an _entity_ here, so the keys are **base** URIs. That means that we could potentially have multiple versions of the type that matches.

I suspect the best thing to do would be to pick the latest, or to look at the entity's entity type and find the version of the property type used inside that.